src/pybind/mgr/cephadm/module.py

```diff
@@ -3055,8 +3055,27 @@ def set_prometheus_access_info(self, user: str, password: str) -> str:
         self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return 'prometheus credentials updated correctly'
 
+    @handle_orch_error
+    def set_prometheus_cert(self, cert: str) -> str:
+        self.set_store(PrometheusService.PROMETHEUS_CERT_CFG_KEY, cert)
+        return 'prometheus cert stored correctly'
+
+    @handle_orch_error
+    def get_prometheus_cert(self) -> str:
+        prometheus_cert = self.get_store(PrometheusService.PROMETHEUS_CERT_CFG_KEY)
+        return prometheus_cert
+
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
+        if url.startswith("http://") or url.startswith("https://"):
+            return f"Invalid URL '{url}'. It should be in the format host_ip:port"
+        try:
+            host, port = url.rsplit(":", 1)
```

Comment:
this will break if the provided IP is ipv6 ...  I'd recommend using the urlparse library and avoid any kind of manual parsing.